converter/internal/prometheusconvert/scrape.go

```diff
@@ -68,7 +69,9 @@ func getScrapeTargets(staticConfig prom_discovery.StaticConfig) []discovery.Targ
 		for _, labelSet := range target.Targets {
 			for labelName, labelValue := range labelSet {
 				targetMap[string(labelName)] = string(labelValue)
-				targets = append(targets, targetMap)
+				newMap := map[string]string{}
+				maps.Copy(newMap, targetMap)
+				targets = append(targets, newMap)
```

Comment:
This was a subtle bug where the `targetMap` was being mutated. We need a copy here. This is why some test cases have the `__address__` label changed. I believe this is the correct behaviour now.